src/System.Management.Automation/engine/remoting/fanin/PriorityCollection.cs

```diff
@@ -220,21 +220,29 @@ internal void Clear()
             lock (_readSyncObject)
             {
                 priorityType = DataPriorityType.Default;
-
-                // send data from which ever stream that has data directly.
+                // Send data from which ever stream that has data directly.
                 byte[] result = null;
-                result = _dataToBeSent[(int)DataPriorityType.PromptResponse].ReadOrRegisterCallback(_onSendCollectionDataAvailable);
-                priorityType = DataPriorityType.PromptResponse;
+                SerializedDataStream promptDataToBeSent = _dataToBeSent[(int)DataPriorityType.PromptResponse];
+                if (promptDataToBeSent is not null)
+                {
+                    result = promptDataToBeSent.ReadOrRegisterCallback(_onSendCollectionDataAvailable);
```

Comment:
I think `SerializedDataStream.ReadOrRegisterCallback` should have a check on `_isDisposed` and fast return in case the instance is disposed, so that we can avoid the possibility of returning a `byte[]` object with all its element being value `0`. Please see my analysis above.

Reply:
My inclination is to make as minimal changes as possible in this code.  The race error occurs during command complete, and whether it is null reference or disposed exception can only reasonably be ignored at this point.  So rather than trying to predict everywhere an error might occur for this race, I would instead just add a try/catch.  But I prefer not to do this yet, and instead just start with these small changes.  We can always add try/catch later, but I am hoping this change will be sufficient, as testing seems to indicate.

Reply:
The possible rase condition issue I described in https://github.com/PowerShell/PowerShell/pull/16683#discussion_r784211992 does not involve a `ObjectDisposed` exception to be thrown, so a try/catch won't help if that issue does happen.

Reply:
Yes, I am suggesting that we ignore that possible state because we are at command complete/clean-up state, but can revisit if it is shown to be a problem.  Thanks for looking at this!